dagmc_stats/DagmcQuery.py

```diff
@@ -160,79 +199,317 @@ def __update_vert_data(self, new_data):
+        tri_vert_index = 0
+
+        for tri in self.tris:
+            side_lengths = self.get_tri_side_length(tri)  # {vert: side_length}
+            side_length_sum_sq_half = sum(map(lambda i: i**2,
+                                          side_lengths.values())) / 2.
+            side_length_prod = np.prod(list(side_lengths.values()))
+            verts = list(self.dagmc_file._my_moab_core.get_adjacencies(
+                tri, 0, op_type=1))
+            for vert_i in verts:
+                side_i = side_lengths[vert_i]
+                d_i = np.arccos((side_length_sum_sq_half -
+                                (side_i**2)) * side_i / side_length_prod)
+                tri_vert_entry = np.array((tri, vert_i, d_i, side_i),
+                               dtype=tri_vert_struct)
+                self._tri_vert_data[tri_vert_index] = tri_vert_entry
+                tri_vert_index += 1
+
+    def __gaussian_curvature(self, vert_i):
+        """Get gaussian curvature value of a vertex
+        Reference: https://www.sciencedirect.com/science/article/pii/
+        S0097849312001203
+        Formula 1
+
+        inputs
+        ------
+            vert_i : vertex entity handle
+
+        outputs
+        -------
+            gc : gaussian curvature value of the vertex
+        """
+        vert_entries = self._tri_vert_data[self._tri_vert_data['vert'] == vert_i]
+        sum_alpha_angles = sum(vert_entries['angle'])
+        gc = np.abs(2 * np.pi - sum_alpha_angles)
+        return gc
+
+    def __get_lri(self, vert_i, gc_all):
+        """Get local roughness value of a vertex
+        Reference: https://www.sciencedirect.com/science/article/pii/
+        S0097849312001203
+        Formula 2, 3
+
+        inputs
+        ------
+            vert_i : vertex entity handle
+            gc_all : dictionary in the form of vertex : gaussian curvature
+                value of the vertex
+
+        outputs
+        -------
+            Lri : local roughness value of the vertex
+        """
+        DIJgc_sum = 0
+        Dii_sum = 0
+        adj_tris = self.dagmc_file._my_moab_core.get_adjacencies(
+                vert_i, 2, op_type=0)
+        adj_tris = list(set(adj_tris) & set(self.tris))
+        vert_j_list = list(self.dagmc_file._my_moab_core.get_adjacencies(
+            adj_tris, 0, op_type=1))
+        vert_j_list.remove(vert_i)
+        for vert_j in vert_j_list:
+            # get tri_ij_list (the list of the two triangles connected to both
+            # vert_i and vert_j)
+            tri_j_list = self.dagmc_file._my_moab_core.get_adjacencies(
+                vert_j, 2, op_type=0)
+            tri_ij_list = list(set(adj_tris) & set(tri_j_list))
+            # rows with tri value as tri_ij_list[0] or tri_ij_list[1]
+            select_tris = (self._tri_vert_data['tri'] == tri_ij_list[0]) | \
+                          (self._tri_vert_data['tri'] == tri_ij_list[1])
```

Comment:
I was using this tool again this week and discovered another slight bug. Just like with the beta angles below, when we have edge vertices/triangles, we need to use the `-1` index because there is only one triangle.